Lib/unittest/case.py

```diff
@@ -857,22 +857,28 @@ def assertAlmostEqual(self, first, second, places=None, msg=None,
             raise TypeError("specify delta or places not both")
 
         if delta is not None:
-            if abs(first - second) <= delta:
+            diff = abs(first - second)
+            if diff <= delta:
                 return
 
-            standardMsg = '%s != %s within %s delta' % (safe_repr(first),
-                                                        safe_repr(second),
-                                                        safe_repr(delta))
+            standardMsg = '%s != %s within %s delta (%s difference)' % (
+                safe_repr(first),
+                safe_repr(second),
+                safe_repr(delta),
+                safe_repr(diff))
         else:
             if places is None:
                 places = 7
 
-            if round(abs(second-first), places) == 0:
+            diff = abs(second - first)
+            if round(diff, places) == 0:
                 return
 
-            standardMsg = '%s != %s within %r places' % (safe_repr(first),
-                                                          safe_repr(second),
-                                                          places)
+            standardMsg = '%s != %s within %r places (%s difference)' % (
```

Comment:
How this differs from the case when delta is specified?

Actually I don't know what is the use case for `assertNotAlmostEqual()`. It is not used in Python tests (except testing `assertNotAlmostEqual()` itself). I suggest to not change it at all.